manager/src/main/java/io/gingersnapproject/k8s/CacheRuleInformer.java

```diff
@@ -0,0 +1,357 @@
+import org.slf4j.LoggerFactory;
+
+import com.google.protobuf.GeneratedMessageV3;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.protobuf.util.JsonFormat;
+
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.Resource;
+import io.fabric8.kubernetes.client.informers.ResourceEventHandler;
+import io.fabric8.kubernetes.client.informers.SharedIndexInformer;
+import io.gingersnapproject.configuration.Configuration;
+import io.gingersnapproject.configuration.Connector;
+import io.gingersnapproject.configuration.Rule;
+import io.gingersnapproject.k8s.configuration.KubernetesConfiguration;
+import io.gingersnapproject.proto.api.config.v1alpha1.EagerCacheRuleSpec;
+import io.gingersnapproject.proto.api.config.v1alpha1.LazyCacheRuleSpec;
+import io.quarkus.runtime.ShutdownEvent;
+import io.quarkus.runtime.StartupEvent;
+
+@ApplicationScoped
+public class CacheRuleInformer {
+
+   private static final Logger log = LoggerFactory.getLogger(CacheRuleInformer.class);
+
+   @Inject
+   Instance<KubernetesClient> client;
+   @Inject
+   KubernetesConfiguration configuration;
+   @Inject
+   Configuration rulesConfig;
+   private SharedIndexInformer<ConfigMap> lazyInformer;
+   private SharedIndexInformer<ConfigMap> eagerInformer;
+
+   void startWatching(@Observes StartupEvent ignore) {
+      log.info("startWatching(): begin");
+      if (client.isUnsatisfied()) {
+         log.info("Kubernetes client not found, not watching config map");
+         return;
+      }
+
+      // TODO: fix logic to handle lazy and eager indep
+      if (configuration.lazyConfigMapName().isEmpty()) {
+         log.info("Lazy config map not specified, not watching config map");
+         return;
+      }
+
+      if (configuration.eagerConfigMapName().isEmpty()) {
+         log.info("Eager config map not specified, not watching config map");
+         return;
+      }
+
+      String lazyRuleName = configuration.lazyConfigMapName().get();
+      String eagerRuleName = configuration.eagerConfigMapName().get();
+      log.info("Informer on lazy rules {} and eager {}", lazyRuleName, eagerRuleName);
+      var RESYNC_PERIOD = 60 * 1000L;
+      KubernetesClient kc = client.get();
+      Resource<ConfigMap> lazyCm = kc.configMaps().withName(lazyRuleName);
+      Resource<ConfigMap> eagerCm = kc.configMaps().withName(eagerRuleName);
+
+      ConfigMapEventHandler lazyHandler = new ConfigMapEventHandler(rulesConfig,
+      LazyCacheRuleSpec.getDefaultInstance());
+      ConfigMapEventHandler eagerHandler = new ConfigMapEventHandler(rulesConfig,
+      EagerCacheRuleSpec.getDefaultInstance());
+
+      // Start informers
+      lazyInformer = lazyCm.inform(lazyHandler, RESYNC_PERIOD);
+      lazyInformer.start();
+      eagerInformer = eagerCm.inform(eagerHandler, RESYNC_PERIOD);
+      eagerInformer.start();
+   }
```

Comment:
The logic for the two informers is almost identical here. It would be good if we could abstract this to avoid duplication.